ios/Exponent/Versioned/Modules/Api/Permissions/EXLocalNotificationRequester.m

```diff
@@ -0,0 +1,42 @@
+// Copyright 2016-present 650 Industries. All rights reserved.
+
+#import "EXLocalNotificationRequester.h"
+#import "RCTUtils.h"
+
+@interface EXLocalNotificationRequester ()
+
+@property (nonatomic, weak) id<EXPermissionRequesterDelegate> delegate;
+
+@end
+
+@implementation EXLocalNotificationRequester
+
++ (NSDictionary *)permissions
+{
+  UIUserNotificationSettings *currentSettings = RCTSharedApplication().currentUserNotificationSettings;
+  
+  EXPermissionStatus status = currentSettings != UIUserNotificationTypeNone
+    ? EXPermissionStatusGranted
+    : EXPermissionStatusUndetermined;
+  
+  return @{
+    @"status": [EXPermissions permissionStringForStatus:status],
+    @"expires": EXPermissionExpiresNever,
+  };
+}
+
+- (void)requestPermissionsWithResolver:(RCTPromiseResolveBlock)resolve rejecter:(__unused RCTPromiseRejectBlock)reject
+{
+  UIUserNotificationType types = UIUserNotificationTypeBadge | UIUserNotificationTypeSound | UIUserNotificationTypeAlert;
+  [RCTSharedApplication() registerUserNotificationSettings:[UIUserNotificationSettings settingsForTypes:types categories:nil]];
+  
+  resolve([[self class] permissions]);
```

Comment:
Can't resolve here - the system may have shown a dialog to the user and we're waiting for them to decide whether to grant permissions. You need to listen to the callback methods on UIApplicationDelegate, then resolve/reject after taht. See `EXRemoteNotificationRequester` for an example of this pattern.